py/src/braintrust/logger.py

```diff
@@ -1163,12 +1173,29 @@ def fetched_data(self):
 
     def _refetch(self):
         if self._fetched_data is None:
-            resp = _state.log_conn().get(
-                f"object/{self.object_type}", params={"id": self.id, "fmt": "json2", "version": self._pinned_version}
-            )
-            response_raise_for_status(resp)
+            data = None
+            try:
+                resp = _state.log_conn().get(
+                        f"object3/{self.object_type}", params={"id": self.id, "fmt": "json2", "version": self._pinned_version, "api_version": 2}
+                )
+                response_raise_for_status(resp)
+                data = resp.json()
+            except Exception as e:
+                # When hitting old versions of the API where the "object3/" endpoint isn't available, fall back to
```

Comment:
add DEPRECATION_NOTICE to this comment. It's not a perfect convention but it's convenient to grep for to find stuff like this.